src/org/opendatakit/briefcase/util/ExportToCsv.java

```diff
@@ -306,6 +308,58 @@ private String getSubmissionValue(EncryptionInformation ei, TreeElement model, E
     return rawElement;
   }
 
+  private String fileHash(File binaryFile) {
+    String fileLocation = binaryFile.getAbsolutePath();
+
+    if (fileHashMap.containsKey(fileLocation))
+      return fileHashMap.get(fileLocation);
+
+    MessageDigest md = null;
+    try {
+      md = MessageDigest.getInstance("SHA1");
+    }
+    catch (NoSuchAlgorithmException e) {
+      e.printStackTrace();
+      return null;
+    }
+
+    FileInputStream fis = null;
+
+    try {
+      fis = new FileInputStream(fileLocation);
+    }
+    catch (FileNotFoundException e) {
+      e.printStackTrace();
+      return null;
+    }
+
+    byte[] dataBytes = new byte[1024];
+
+    int nread = 0;
+    try {
+      while ((nread = fis.read(dataBytes)) != -1) {
+        md.update(dataBytes, 0, nread);
+      }
+    }
+    catch (IOException e) {
+      e.printStackTrace();
+      return null;
+    }
+
+    byte[] mdbytes = md.digest();
+
+    //convert the byte to hex
+    StringBuffer sb = new StringBuffer();
+    for (int i = 0; i < mdbytes.length; i++) {
+      sb.append(Integer.toString((mdbytes[i] & 0xff) + 0x100, 16).substring(1));
+    }
+
+    String hashVal = sb.toString();
+    fileHashMap.put(fileLocation, hashVal);
+
+    return hashVal;
+  }

```

Comment:
@shivam-tripathi There is already a method in briefcase/util known as getMD5hash that returns the MD5 hash of a file. i think you should make use of that. Cheers